homeassistant/components/nibe_heatpump/water_heater.py

```diff
@@ -0,0 +1,182 @@
+        if address := desc.active_accessory:
+            self._coil_active_accessory = _get(address)
+
+        self._coil_hot_water_comfort_mode = _get(desc.hot_water_comfort_mode)
+
+        if self._coil_temporary_lux:
+            if (
+                self._coil_temporary_lux.reverse_mappings
+                and VALUES_TEMPORARY_LUX_ONE_TIME_INCREASE
+                in self._coil_temporary_lux.reverse_mappings
+            ):
+                self._attr_operation_list.append(STATE_HIGH_DEMAND)
+            else:
+                LOGGER.warning("Skipping temporary lux since no mappings exist")
+
+        self._attr_temperature_unit = self._coil_current.unit
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        if not self.coordinator.data:
+            return
+
+        def _get_float(coil: Coil | None) -> float | None:
+            if coil is None:
+                return None
+            return self.coordinator.get_coil_float(coil)
+
+        def _get_value(coil: Coil | None) -> int | str | float | None:
+            if coil is None:
+                return None
+            return self.coordinator.get_coil_value(coil)
+
+        self._attr_current_temperature = _get_float(self._coil_current)
+
+        if (mode := _get_value(self._coil_hot_water_comfort_mode)) and isinstance(
+            mode, str
+        ):
+            self._attr_target_temperature_low = _get_float(
+                self._coil_start_temperature.get(mode)
+            )
+            self._attr_target_temperature_high = _get_float(
+                self._coil_stop_temperature.get(mode)
+            )
+        else:
+            self._attr_target_temperature_low = None
+            self._attr_target_temperature_high = None
+
+        if (
+            _get_value(self._coil_temporary_lux)
+            == VALUES_TEMPORARY_LUX_ONE_TIME_INCREASE
+        ):
+            self._attr_current_operation = STATE_HIGH_DEMAND
+        else:
+            self._attr_current_operation = STATE_HEAT_PUMP
+
+        self.async_write_ha_state()
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        if not self.coordinator.last_update_success:
+            return False
+
+        if not self._coil_active_accessory:
+            return True
+
+        if active_accessory := self.coordinator.get_coil_value(
+            self._coil_active_accessory
+        ):
+            return active_accessory == "ON"
+
+        return False
+
+    async def async_set_operation_mode(self, operation_mode: str) -> None:
+        """Set new target operation mode."""
+        if not self._coil_temporary_lux:
+            raise HomeAssistantError("Not supported")
+
+        if operation_mode == STATE_HEAT_PUMP:
+            await self.coordinator.async_write_coil(
+                self._coil_temporary_lux, VALUES_TEMPORARY_LUX_INACTIVE
+            )
+        elif operation_mode == STATE_HIGH_DEMAND:
```

Comment:
Should this raise before attempting the write if `STATE_HIGH_DEMAND` isn't a valid operation mode because it was never appended to the list in `__init__`?